File: 22June2024.cpp
long long ExtractNumber(string sentence) {

        // code here
            regex regex("[0-9]+");
            smatch match;
            string s = sentence;
            vector<string> numbers;
    
            while (regex_search(s, match, regex)) {
                numbers.push_back(match.str());
                s = match.suffix().str();
            }
    
            
            long long max = LLONG_MIN;
    
            for (const auto& num : numbers) {
                if (num.find('9') == string::npos) {
                    long long numValue = stoll(num);
                    if (max < numValue) {
            
                        max = numValue;
                    }
                }
            }
    
            return max == LLONG_MIN ? -1 : max;
        }

File: 22June2024.java
long ExtractNumber(String sentence) {
        // code here
        Pattern pattern = Pattern.compile("[0-9]+");
        Matcher matcher = pattern.matcher(sentence);
        List<String> nums = new ArrayList<>();
        while(matcher.find()){
            nums.add(matcher.group());
        }
        
        long max = Long.MIN_VALUE;
        for(String numStr : nums){
            if(!numStr.contains("9")){
                long num = Long.parseLong(numStr);
                max = Math.max(max, num);
            }
        }        
        return max == Long.MIN_VALUE ? -1 : max;
    }
